docs/templates.md

```diff
@@ -1,6 +1,36 @@
 # Headers and Decorators
 
 ## Template
+### Templates using decorators
+#### The usage
+```python
+from pyccel.decorators import template
+@template(name='T', types=['int','real'])
+def f(a : 'T', b : 'T'):
+	pass
+```
+Arguments:
+-   name: the name of the template
+-   types: the types the template represents.
+---
+*Note:*
+The arguments **name** and **types** could also be passed of the form
+`@template(T, types=['int', 'real'])` without specifying the keywords.and the order in which the decorators are called is not important.
+
+---
+When  a function is decorated with the template decorator:
+-   The templates are only available to the decorated function.
+-   The templates overrides any existing templates with the same name (declared as header comment).
```

Comment:
Typo: overrides -> override.

But maybe a better sentence is "A template overrides any existing template with the same name [...]."

Reply:
Fixed 9934fdf8